pymc/model_graph.py

```diff
@@ -102,11 +102,42 @@ def get_parents(self, var: TensorVariable) -> Set[VarName]:
         parents = self._get_ancestors(var, func)
         return self._filter_parents(var, parents)
 
-    def make_compute_graph(self) -> Dict[str, Set[VarName]]:
+    def vars_to_plot(self, selected_vars=None):
+        if selected_vars is None:
+            return self.var_names
+        else:
+            selected_vars = set(selected_vars)
+
+            for var_name in selected_vars:
+                if var_name not in self.var_names:
+                    raise ValueError(f"{var_name} is not part of the defined model.")
+
+                for model_var in self.var_list:
+                    if hasattr(model_var.tag, "observations"):
+                        if model_var.tag.observations == self.model[var_name]:
+                            selected_vars = selected_vars.union({model_var.name})
+
+            selected_ancestors = set(
+                filter(
+                    lambda rv: rv.name in self.var_names,
+                    list(ancestors([self.model[var_name] for var_name in selected_vars])),
+                )
+            )
+
+            for var in selected_ancestors:
+                if hasattr(var.tag, "observations"):
+                    selected_ancestors = selected_ancestors.union({var.tag.observations})
+
+            selected_ancestors = [var.name for var in selected_ancestors]
+
+            # ordering of self.var_names is important
+            return selected_ancestors
```

Comment:
Is is sorted anywhere?
Does the return value need to be sorted?

And you can return the listcomprehension without creating another local variable

Reply:
Not sure about sorted, would have to revisit why I wrote this comment again